### PR TITLE
BUG: a few more small bug fixes

### LINKED ISSUE #8726: BUG: out-of-sample prediction with nan values get unexpected result
This is an OLS regression model from [this page](https://www.statsmodels.org/stable/example_formulas.html):
```
import statsmodels.api as sm
import statsmodels.formula.api as smf
import numpy as np
import pandas as pd

# example from https://www.statsmodels.org/stable/example_formulas.html
df = sm.datasets.get_rdataset("Guerry", "HistData").data
df = df[['Lottery', 'Literacy', 'Wealth', 'Region']].dropna()
mod = smf.ols(formula='Lottery ~ Literacy', data=df)
res = mod.fit()
```
When this model is fitted, we can use this model to predict: 

```
# out of sample prediction without NaN values. This works as expected.
res.predict(pd.Series([0,1,2,3], name='Literacy'))
```

But when we have NaN values in the exog data, the `predict()` method returns unexpected result, which is a pandas series with only NaN value indexed by None:
 
```
# ****out of sample prediction with NaN values. This gives a strange result!****
res.predict(pd.Series([0,1,2,None,3], name='Literacy'))
```

statsmodels: 0.13.5
pandas: 1.4.4
numpy: 1.23.5
python:3.9.16

### LINKED ISSUE #8729: ENH/BUG: allow fit keyword ridge_factor in newton
`model.fit(method="newton", ridge_factor= 1e-5)`

issues a warning about unused kwarg, and ridge_factor is not transmitted to the newton optimizer.
I wanted to try out fit with a large ridge factor to force convergence (to a possibly regularized solution)

I'm not sure whether this ever worked or was a refactoring victim.

AFAICS, it's because ridge_factor is explicit keyword and not in `kwargs` in signature, base.optimizer
```
        def _fit_newton(f, score, start_params, fargs, kwargs, disp=True,
                maxiter=100, callback=None, retall=False,
                full_output=True, hess=None, ridge_factor=1e-10)
```

### PR DESCRIPTION
**Pull request by josef-pkt**
- catch Convergence warning when computing start_params. I had missed  few
  https://stats.stackexchange.com/questions/608609/statsmodels-zeroinflatedpoisson-unable-to-converge 
- closes #8726

I still have 2 or 3 uncommitted changes to the code, not sure yet about which way to fix
#8729
manova skip intercept https://github.com/statsmodels/statsmodels/issues/8713#issuecomment-1458655396

### CODE REVIEW

**Comment by josef-pkt** [comment]
unit tests for this PR look fine (I forgot to add a seed to the random test)

unit tests fail on many environment, AFAICS, all that use pandas 2.0.0
